iodata/formats/qchemlog.py

```diff
@@ -366,3 +397,71 @@ def _helper_thermo(lit: LineIterator) -> Tuple:
+    """Load Energy decomposition information"""
+    next(lit)
+    eda2_dic = {}
+    for line in lit:
+        if line.startswith('  Orthogonal Fragment Subspace Decomposition'):
+            next(lit)
+            for line_2 in lit:
+                if line_2.startswith('     --------------------'):
+                    break
+                else:
+                    info = line_2.strip().split()
+                    if info[0] == 'E_elec':
+                        eda2_dic[info[0].lower()] = float(info[4])
+                    elif info[0] == 'E_pauli':
+                        eda2_dic[info[0].lower()] = float(info[4])
+                    elif info[0] == 'E_disp':
+                        eda2_dic[info[0].lower()] = float(info[4])
+        elif line.startswith('     Terms summing to E_pauli'):
+            next(lit)
+            pauli = []
+            for line_2 in lit:
+                if line_2.startswith('  --------------------'):
+                    break
+                else:
+                    info = line_2.strip().split()
+                    pauli.append(float(info[3]))
+            eda2_dic['e_pauli'] = np.array(pauli)
+        elif line.startswith('  Classical Frozen Decomposition'):
+            next(lit)
+            for line_2 in lit:
+                if line_2.startswith('  --------------------'):
+                    break
+                else:
+                    info = line_2.strip().split()
+                    if info[0] == 'E_cls_elec':
+                        eda2_dic[info[0].lower()] = float(info[5])
+                    elif info[0] == 'E_cls_pauli':
+                        eda2_dic[info[0].lower()] = float(info[5])
+                    elif info[0].split("[")[1] == 'E_mod_pauli':
+                        eda2_dic[info[0].split("[")[1].lower()] = float(info[5])
+        elif line.startswith('Simplified EDA Summary'):
+            next(lit)
+            for line_2 in lit:
+                if line_2.startswith('--------------------'):
+                    break
+                else:
+                    info = line_2.strip().split()
+                    if info[0] == 'PREPARATION':
+                        eda2_dic[info[0].lower()] = float(info[1])
+                    elif info[0] == 'FROZEN':
+                        eda2_dic[info[0].lower()] = float(info[1])
+                    elif info[0].split("[")[-1] == 'PAULI':
+                        eda2_dic[info[0].split("[")[-1].lower()] = float(info[1].split("]")[0])
+                    if info[0] == 'DISPERSION':
+                        eda2_dic[info[0].lower()] = float(info[1])
+                    elif info[0] == 'POLARIZATION':
+                        eda2_dic[info[0].lower()] = float(info[1])
+                    elif info[0] == 'CHARGE':
+                        eda2_dic[info[0].lower() + ' ' + info[1].lower()] = float(info[2])
+                    elif info[0] == 'TOTAL':
+                        eda2_dic[info[0].lower()] = float(info[1])
```

Comment:
I think the same type of change (mentioned above), can make this shorter...

Reply:
I couldn't do it for all `elif` sections because some are specific for that particular component but I think it reduced length quite a bit. Thanks for the suggestion!